task.ps1

```diff
@@ -35,5 +43,6 @@ for (($zone = 1); ($zone -le 2); ($zone++) ) {
     -SubnetName $subnetName `
     -VirtualNetworkName $virtualNetworkName `
     -SecurityGroupName $networkSecurityGroupName `
```

Comment:
The parameter `-SecurityGroupName` in `New-AzVm` might not be valid. Typically, you would associate a network security group with a network interface or subnet, not directly with a VM. Consider reviewing the documentation for the correct usage.